### DISCUSSION TITLE
Suppress Ulauncher when certain apps are active

### DISCUSSION THREAD

**Discussion by marquesarthur** [Ulauncher/Ulauncher#890] [Q&A] [answered]
## Question

Having recently migrated from macOS to Linux Ubuntu 18.04, I am in love with Ulauncher.
However, I hit a block functionality when coding on Pycharm or VSCode.
The autocomplete shortcut ctrl+space conflicts with my setup Ulaucher shortcut. 
On macOS, there are ways to disable launcher functionalities when certain apps are the active window. Is there a way to do so using Ulaucher? I really like the ctrl + space shortcut and I don't want to change that.

**Comment by friday** [comment]
No there isn't. You can solve the problem in these ways:
* Change the key combo in either one
* If you're technically inclined, create a wrapper for ulauncher-toggle that triggers conditionally depending on the [active application](https://superuser.com/questions/382616/detecting-currently-active-window), then bind the hotkey in your desktop environment settings (see https://github.com/Ulauncher/Ulauncher/discussions/1347).

**Comment by valkedev** [comment, marked as answer]
My tool for excluding specific windows:
```bash
#!/usr/bin/bash
excluded=('Minecraft')
currentWindow=`xdotool getwindowfocus getwindowname`
ulauncherWindowTitle='Ulauncher - Application Launcher'

for excludedApplication in "${excluded[@]}"; do
  if [[ "$excludedApplication" = "$currentWindow" ]]; then
    exit 1
  fi
done

$(ulauncher)
$(xdotool  windowactivate `xdotool search --onlyvisible --name "${ulauncherWindowTitle}"`)
```

You should install xdotool first: `sudo apt install xdotool`
Window titles are listed in `excluded` array
Just set shortcut to this app.

EDIT: Fixed bug when Ulauncher window became unactive when triggered
